tests/integration/test_private_repos.py

```diff
@@ -0,0 +1,63 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import os
+
+import pytest
+from git import Repo
+
+from . import utils
+
+
+@pytest.mark.parametrize("env_package", ["https", "ssh"])
+def test_private_repos(env_package, test_env, tmpdir):
+    """
+    Validate a cachito request with no package managers to a private repo.
+
+    Process:
+    Send new request to the Cachito API
+    Send request to check status of existing request
+
+    Checks:
+    * Check that the request completes successfully
+    * Check that no packages are identified in response
+    * Check that no dependencies are identified in response
+    * Check that the source tarball includes the application source code
```

Comment:
Thanks for adding the docstring :tada: 

Could we also mention how we check the application source code here, and why we have to do it differently than the other tests?